source/fundamentals/logging.txt

```diff
@@ -0,0 +1,378 @@
+      `the Logback Manual <https://logback.qos.ch/manual/configuration.html>`__.
+
+      .. tabs::
+
+   .. tab:: Log4j2
+      :tabid: Log4j2-binding
+
+      Specify Log4j2 configurations in a file named ``log4j2.xml``. Your
+      ``log4j2.xml`` file does not have to be in a specific location, but it must
+      be accessible from your classpath. 
+
+      The Log4j2 framework defines the following log levels. You can see the
+      log levels listed below in order from most urgent to least urgent: 
+      
+      - OFF
+      - FATAL
+      - ERROR
+      - WARN
+      - INFO
+      - DEBUG
+      - TRACE
+      - ALL 
+
+      Set your ``log4j2.xml`` file to the following.
+
+      .. code-block:: xml
+
+         <?xml version="1.0" encoding="UTF-8"?>
+         <Configuration status="INFO">
+            <Appenders>
+               <Console name="Console" target="SYSTEM_OUT">
+                     <PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
+               </Console>
+            </Appenders>
+            <Loggers>
+               <Root level="info">
+                     <AppenderRef ref="Console"/>
+               </Root>
+            </Loggers>
+         </Configuration>
+
+      To test that your logger configuration was successful, run the following
+      code.
+
+      .. include:: /includes/fundamentals/logging.rst
+
+      You should see output that resembles the following.
+
+      .. code-block:: none
+
+         ...
+         10:14:57.633 [cluster-ClusterId{value=<your cluster id>, description='null'}-<your connection uri>] INFO  org.mongodb.driver.cluster - Discovered replica set primary <your connection uri>
+         10:14:57.790 [main] INFO  org.mongodb.driver.connection - Opened connection [connectionId{localValue:7, serverValue:<your server value>}] to <your connection uri>
+      
+      For more information on configuring Log4j2, see the official 
+      `Log4j2 configuration guide
+      <https://logging.apache.org/log4j/2.x/manual/configuration.html>`__.
+
+Logger Names
+------------
+
+Your logger uses logger names to help organize different logging events. Logger
+names are strings that form a hierarchy. A logger is an ancestor of another logger if
+its name followed by a ``"."`` is a prefix of the other logger's name. For example,
+``"grandparent"`` is an ancestor of ``"grandparent.parent"`` which is an
+ancestor of ``"grandparent.parent.child"``.
+
+For a concrete example, this is what a logger hierarchy looks like in code.
+
+.. code-block:: java
+
+   import org.slf4j.Logger;
+   import org.slf4j.LoggerFactory;
+
+   ...
+
+   Logger logger_parent = LoggerFactory.getLogger("parent");
+   Logger logger_child = LoggerFactory.getLogger("parent.child");
+
+A logger inherits the properties of its ancestor logger and can define
+its own. You can think of this as similar to class inheritance in Java. 
+
+The makers of the MongoDB Java driver have defined the following logger names to
```

Comment:
Thanks for the note. Fix: 

For background around this phrasing (feel free to ignore this comment and resolve if you are not interested): when I was studying logger names, I had trouble telling where logger names were getting defined (logging framework, people using logging framework, or SLF4J). I said "the makers of the mongodb java driver" to try and put a face to which involved party made the following list. In my head that was easier to understand, but in hindsight it does deviate from the majority of our content (I agree that it would apply to everywhere else in the driver), and I agree that taking it out is beneficial.